feat: add `math/base/special/cpolarf`

anandkaranubc opened the pull request:
Progresses https://github.com/stdlib-js/stdlib/issues/649

## Description

> What is the purpose of this pull request?

This pull request:

-  Adds the complete implementation for `math/base/special/cpolarf`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   Progresses https://github.com/stdlib-js/stdlib/issues/649

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

anandkaranubc committed:
feat: temp commit

anandkaranubc committed:
Merge branch 'develop' into feat/cpolarf

anandkaranubc committed:
feat: add `math/base/special/cpolarf`

anandkaranubc committed:
Merge branch 'develop' into feat/cpolarf

anandkaranubc committed:
bench: use `random_uniform`

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
docs: fix examples

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
Merge branch 'develop' into feat/cpolarf

anandkaranubc committed:
chore: minor clean-up

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
Merge branch 'develop' into feat/cpolarf

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/cpolarf/examples/index.js:
```diff
@@ -0,0 +1,38 @@
+	z = new Complex64( re, im );
+	o = cpolarf( z );
+	z = z.toString();
+	console.log( 'absf(%s) = %d. argf(%s) = %d', z, o[ 0 ], z, o[ 1 ] );
```
The reason why it doesn't have the same words as `cpolarf`, `cphasef` and `cabsf` in here and in README, is because they cause a cs-spell-checker lint error.

anandkaranubc replied:
As compared to the `example.c` file

Planeshifter replied:
The lint rule's behavior is configured in https://github.com/stdlib-js/stdlib/blob/0da78b462b0783c2d7d145d477b20c8750ed8486/etc/eslint/rules/spellcheck.js. Would be good to add exceptions there.

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/cpolarf/README.md:
````diff
@@ -0,0 +1,228 @@
+-   **cphasef**: `[out] float*` destination for the phase value in radians.
+
+```c
+float stdlib_base_cpolarf( const stdlib_complex64_t z, float *cabsf, float *cphasef );
````
```suggestion
void stdlib_base_cpolarf( const stdlib_complex64_t z, float *cabsf, float *cphasef );
```

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/cpolarf/docs/types/index.d.ts:
```diff
@@ -0,0 +1,85 @@
+	* var v = cpolarf( new Complex64( 5.0, 3.0 ) );
+	* // returns [ ~5.83, ~0.5404 ]
+	*/
+	( z: Complex64 ): Array<number>;
```
Same question of whether we should return a tuple here, which would be slightly more type-safe:

```suggestion
	( z: Complex64 ):  [ number, number ];
```

Planeshifter reviewed (changes requested):
Should be ready once updating return type and fixing function signature.

anandkaranubc committed:
Merge branch 'develop' into feat/cpolarf

anandkaranubc committed:
chore: apply suggestions from review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/cpolarf/examples/index.js:
```diff
@@ -0,0 +1,30 @@
+
+'use strict';
+
+var Complex64Array = require( '@stdlib/array/complex64' );
+var uniform = require( '@stdlib/random/array/uniform' );
+var logEachMap = require( '@stdlib/console/log-each-map' );
+var cpolarf = require( './../lib' );
+
+// Create an array of random numbers:
+var arr = new Complex64Array( uniform( 200, -100.0, 100.0 ) );
+
+// Compute the polar form of each number in the array:
+logEachMap( 'cpolarf(%s) = [%s]', arr, cpolarf );
```
@Planeshifter If this refactoring looks good, I will also apply that to `cpolar`. 

One caveat: it is slightly different from the C examples.

Planeshifter replied:
Yeah, looks fine to me!

Planeshifter reviewed (approved):
Ready to merge!
